docs/source/hyper-parameter-search.rst

```diff
@@ -1,33 +1,232 @@
+
+1. ":ref:`hyperparameter.drop-in`" will mention classes that are drop-in
+   replacements for Scikit-learn but work nicely with Dask objects.
+2. ":ref:`hyperparameter.incremental`" will mention classes that work well with
+   large datasets.
+3. ":ref:`hyperparameter.adaptive`" will mention classes that avoid extra
+   computation and find high-performing hyperparameters more quickly.
+
+.. _hyperparameter.scaling:
+
+Scaling hyperparameter searches
+-------------------------------
+
+Dask-ML provides classes to avoid the two most common issues in hyperparameter
+optimization, when the hyperparameter search is...
+
+1. "**memory constrained"**. This happens when the dataset size is too large to
+   fit in memory.  This typically happens when a model needs to be tuned for a
+   larger-than-memory dataset after local development.
+2. "**compute constrained**". This happen when the computation takes too long
+   even with data that can fit in memory.  This typically happens when many
+   hyperparameters need to be tuned, which especially common with neural
+   networks.
+
+"Memory constrained" searches happen when the data doesn't fit in the memory of
+a single machine:
+
+.. code:: python
+
+   # not memory constrained
+   import pandas as pd
+   df = pd.read_csv("data/0.parquet")
+   print(df.shape)  # (30000, 200) => 23MB
+
+   # memory constrained
+   import dask.dataframe as dd
+   # Read 1000 of the above dataframes (22GB of data)
+   ddf = dd.read_parquet("data/*.parquet")
+
+"Compute constrained" is when the hyperparameter search takes too long even if
+the data fits in memory. There might a lot of hyperparameters to search, or the
+model may require specialized hardware like GPUs:
+
+.. code:: python
+
+   import pandas as pd
+   from scipy.stats import uniform, loguniform
+   from sklearn.linear_model import SGDClasifier
+
+   df = pd.read_parquet("data/0.parquet")  # data to train on; 23MB as above
+
+   model = SGDClasifier()
+
+   # not compute constrained
+   params = {"l1_ratio": uniform(0, 1)}
+
+   # compute constrained
+   params = {
+       "l1_ratio": uniform(0, 1),
+       "alpha": loguniform(1e-5, 1e-1),
+       "penalty": ["l2", "l1", "elasticnet"],
+       "learning_rate": ["invscaling", "adaptive"],
+       "power_t": uniform(0, 1),
+       "average": [True, False],
+   }
+
+These issues are independent and both can happen the same time. Dask-ML has
+tools to address all 4 combinations. Let's look at each case.
+
+Neither compute nor memory constrained
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This case happens when there aren't many hyperparameters to tune and the data
+fits in memory. This is common when the search doesn't take too long to run.
+
+Scikit-learn can handle this case:
+
+.. autosummary::
+   sklearn.model_selection.GridSearchCV
+   sklearn.model_selection.RandomizedSearchCV
+
+Dask-ML also has some drop in replacements for the Scikit-learn versions that
+works well with Dask.
```

Comment:
```suggestion
works well with Dask collections.
```

and maybe link to https://docs.dask.org/en/latest/user-interfaces.html#high-level-collections